pocs/tests/test_constraints.py

```diff
@@ -229,3 +233,28 @@ def test_moon_avoidance(observer):
 
     assert veto1 is False and veto2 is False
     assert score2 > score1
+
+
+def test_already_visited(observer):
+    avc = AlreadyVisited()
+
+    time = Time('2016-08-13 10:00:00')
+
+    observation1 = Observation(Field('HD189733', '20h00m43.7135s +22d42m39.0645s'))  # HD189733
+    observation2 = Observation(Field('Hat-P-16', '00h38m17.59s +42d27m47.2s'))  # Hat-P-16
+    observation3 = Observation(Field('Sabik', '17h10m23s -15d43m30s'))  # Sabik
+
+    observed_list = OrderedDict()
+
+    observation1.seq_time = '01:00'
+    observation2.seq_time = '02:00'
+    observation3.seq_time = '03:00'
+
+    observed_list[observation1.seq_time] = observation1
+    observed_list[observation2.seq_time] = observation2
+
+    veto1, score1 = avc.get_score(time, observer, observation1)
```

Comment:
In `dispatch.py` you [can see](https://github.com/panoptes/POCS/blob/develop/pocs/scheduler/dispatch.py#L62) how common properties is passed to `get_score` by using the `**common_properties` syntax, which passes individual keys of the dictionary to the method (as opposed to passing the dictionary itself).

So in your `get_score` here you just need to pass the `observed_list` with a keyword, so `observed_list=observed_list`. Then the generic `kwargs` in the get_score you wrote will have it available as `kwargs.get('observed_list')`. This is one of the points of using `**kwags` as a parameter, it can pick up keyword arguments that are not otherwise explicitly stated. We have a `get_score` in all of the constraints and not all the keywords apply to all the constraint types so this is how you design for something like that.

Reply:
Ahh, interesting. I wasn't too sure how `**kwargs` worked but that makes plenty of sense. Thanks!